dig.php

```diff
@@ -57,184 +58,63 @@ function sig_handler($signo)
-    }
+    if ($begin >= $prod_langchecker_only) {
+        if ($begin != $today) {
+            print 'Sorry, we can\'t generate data for ' . $prod_langchecker_only->format('Y-m-d') . " and beyond.\n";
+        } else {
+            print "Fetching current JSON from Langchecker\n";
+            $json_day = json_decode(file_get_contents('https://l10n.mozilla-community.org/~pascalc/langchecker/?action=count&json'), true);
 
-    // On 2015-10-14 we reorganized the langchecker app file structure
-    if (is_dir($git_langchecker .'/app/config/')) {
-        copy($repos . '/settings2015-10-13.inc.php', $git_langchecker .'/app/config/settings.inc.php');
-    }
+            if (is_array($json_day)) {
+                ksort($json_day);
+            }
 
-    if (! is_dir($git_langchecker . '/vendor') && is_file($git_langchecker . '/composer.json')) {
-        print "Installing composer dependencies.\n";
-        exec("php {$git_langchecker}/composer.phar install");
-    }
+            if (empty($json_day)) {
+                print "Empty json source\n";
+            }
 
-    if (! isset($composer_sig) && is_file($git_langchecker . '/composer.json')) {
-        $composer_sig = sha1(file_get_contents($git_langchecker . '/composer.json'));
-    }
+            $data[$today->format('Y-m-d')] = $json_day;
 
-    if (isset($composer_sig)) {
-        if (sha1(file_get_contents($git_langchecker . '/composer.json')) != $composer_sig) {
-            print "Updating composer dependencies.\n";
-            exec("php {$git_langchecker}/composer.phar");
-            $composer_sig = sha1(file_get_contents($git_langchecker . '/composer.json'));
-        } else {
-            // Updating autoloader
-            exec("php {$git_langchecker}/composer.phar dump-autoload");
-        }
-    }
-
-    chdir($git_mozorg);
-    exec("git checkout `git rev-list -n 1 --before=\"${vcs_day}\" master`");
-    chdir($svn_misc);
-    exec('svn up -r{"' . $vcs_day . '"}');
-    chdir($git_stores);
-    exec("git checkout `git rev-list -n 1 --before=\"${vcs_day}\" master`");
-
-    // from  2015-10-10 to  2015-10-14 we reorganized the langchecker app file structure
-    if (is_dir($git_langchecker . '/app/classes/') && is_file($git_langchecker . '/web/index.php')) {
-        chdir($git_langchecker . '/web/');
-        if (! $reorg) {
-            killDevelopmentServer($pid_file);
-            startDevelopmentServer($pid_file);
-            $reorg = true;
+            file_put_contents($data_path, json_encode($data, JSON_PRETTY_PRINT));
         }
     } else {
-        $reorg = false;
-        chdir($git_langchecker);
-    }
-
-    print $day . "\n";
-    // Analyse data
-    chdir($app);
-    $json_day = json_decode(file_get_contents('http://localhost:8082/?action=count&json'), true);
-
-    if (is_array($json_day)) {
-        ksort($json_day);
+        include 'legacy_script.php';
     }
+} else {
+    $date_override = false;
```

Comment:
I'm lost: if we don't provide a date to the script, why aren't we fetching the JSON? I'd expect this to be the common way to call the script.